File: JVM/jnetcall-java-tests/src/test/java/jnetcall/java/tests/io/FileTransportTest.java
package jnetcall.java.tests.io;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.concurrent.atomic.AtomicInteger;

import org.javatuples.Pair;

import jnetbase.java.threads.ThreadExecutor;
import jnetcall.java.api.io.ISendTransport;
import jnetcall.java.impl.io.disk.FolderTransport;

public final class FileTransportTest extends TransportTest {

    private final String Folder = "FileTmp";

    private static AtomicInteger _offset = new AtomicInteger();

    private static int getNextOffset() {
        return _offset.incrementAndGet();
    }

    @Override
    protected Pair<ISendTransport, ISendTransport> getBoth() {
        int offset = getNextOffset();
        Path first = Paths.get(Folder, 13001 + offset + "");
        Path second = Paths.get(Folder, 13051 + offset + "");
        ThreadExecutor exe = new ThreadExecutor();
        FolderTransport left = new FolderTransport(
                Encoding,
                first,
                second,
                exe
        );
        FolderTransport right = new FolderTransport(
                Encoding,
                second,
                first,
                exe
        );
        return Pair.with(left, right);
    }

    @Override
    protected int getMaxListWait() { return 20; }
}

File: NET/JNetCall.Sharp.Tests/IO/FileTransportTest.cs
using System.IO;
using System.Threading;
using JNetCall.Sharp.API.IO;
using JNetCall.Sharp.Impl.IO.Disk;

namespace JNetCall.Sharp.Tests.IO
{
    public sealed class FileTransportTest : TransportTest
    {
        private const string Folder = @"FileTmp";

        private static int _offset;
        private static int NextOffset => Interlocked.Increment(ref _offset);

        protected override (ISendTransport, ISendTransport) GetBoth()
        {
            var offset = NextOffset;
            var portOne = 13001 + offset + "";
            var portTwo = 13051 + offset + "";
            var left = new FolderTransport(
                Encoding,
                Path.Combine(Folder, portOne),
                Path.Combine(Folder, portTwo)
            );
            var right = new FolderTransport(
                Encoding,
                Path.Combine(Folder, portTwo),
                Path.Combine(Folder, portOne)
            );
            return (left, right);
        }

        protected override int MaxListWait => 20;
    }
}